specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,289 @@
+    COREWEBVIEW2_BROWSING_DATA_KIND_GENERAL_AUTOFILL | 
+    COREWEBVIEW2_BROWSING_DATA_KIND_PASSWORD_AUTOSAVE |
+    COREWEBVIEW2_BROWSING_DATA_KIND_BROWSING_HISTORY | 
+    COREWEBVIEW2_BROWSING_DATA_KIND_SETTINGS,
+} COREWEBVIEW2_BROWSING_DATA_KIND;
+
+[uuid(DAF8B1F9-276D-410C-B481-58CBADF85C9C), object, pointer_default(unique)]
+interface ICoreWebView2Profile : IUnknown {
+
+  /// Clear browsing data based on a data type. This method takes two parameters, 
+  /// the first being a mask of one or more COREWEBVIEW2_BROWSING_DATAKIND values. OR operation(s) 
+  /// can be applied to multiple COREWEBVIEW2_BROWSING_DATA_KIND values to create a mask 
+  /// representing multiple data types. The browsing data kinds that are supported 
+  /// are listed below. These data kinds follow a hierarchical structure in which 
+  /// nested bullet points are included in their parent bullet point's data kind.
+  /// Ex: DOM storage is encompassed in site data which is encompassed in the profile data. 
+  /// * Profile
+  ///   * Site Data
+  ///     * DOM Storage: App Cache, File Systems, Indexed DB, Local Storage, Web SQL, Cache 
+  ///         Storage
+  ///     * Cookies 
+  ///   * HTTP Cache 
+  ///   * Download History
+  ///   * General Autofill 
+  ///   * Password Autosave
+  ///   * Browsing History
+  ///   * Settings  
+  /// The completed handler will be invoked when the browsing data has been cleared and will 
+  /// indicate if the specified data was properly cleared.
+  /// If the WebView object is closed before the clear browsing data operation 
+  /// has completed, the handler will be released, but not invoked. In this case 
+  /// the clear browsing data operation may or may not be completed. 
+  /// ClearBrowsingData clears the `dataKinds` for all time. 
+  HRESULT ClearBrowsingData(
+      [in] UINT64 dataKinds,
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler);
+  
+  /// ClearBrowsingDataInTimeRange takes in two additional parameters for the 
+  /// start and end time for which it should clear the data between.  The startTime and endTime 
+  /// parameters correspond to the number of seconds since the UNIX epoch. 
+  ///
+  /// \snippet AppWindow.cpp ClearBrowsingData
+  HRESULT ClearBrowsingDataInTimeRange(
+      [in] UINT64 dataKinds, 
+      [in] double startTime,
+      [in] double endTime, 
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler);
+}
+
+/// The caller implements this interface to receive the ClearBrowsingData result.
+[uuid(27676699-FE17-4E2B-8C1B-267395A04ED5), object, pointer_default(unique)]
+interface ICoreWebView2ClearBrowsingDataCompletedHandler : IUnknown {
+  /// Provide the completion status and result of the corresponding
+  /// asynchronous method. The result indicates if the ClearBrowsingData 
+  /// call has completed successfully and all intended data has been cleared. 
+  HRESULT Invoke(
+      [in] HRESULT errorCode, 
+      [in] BOOL isSuccessful);
```

Comment:
What are the error cases and which will result in an HRESULT error code and which will result in a false isSuccessful?

Reply:
There is only the case in which the method call is interrupted before the browsing data is finished clearing, in which isSuccessful is false and the errorCode is S_OK.  I saw this is what we do for TrySuspend.

Reply:
What does it mean that the call is interrupted?

Reply:
Ok, looked through the remover code again and it does not look like there are any known cases of failure.